addons/mail/models/mail_activity.py

```diff
@@ -795,3 +796,132 @@ def activity_unlink(self, act_type_xmlids, user_id=None):
             domain = ['&'] + domain + [('user_id', '=', user_id)]
         self.env['mail.activity'].search(domain).unlink()
         return True
+
+    def _read_progress_bar(self, domain, group_by, progress_bar):
+        if not (progress_bar['field'] == 'activity_state' and self._fields[group_by].store):
+            return super()._read_progress_bar(domain, group_by, progress_bar)
+
+        # use datetime.utcnow() instead of date.today(), because
+        # it's used in _compute_state_from_date and on mocking in tests
+        today_utc = datetime.utcnow()
+        today = date(year=today_utc.year, month=today_utc.month, day=today_utc.day)
+
+        with_alias = 'mail_activity_min'
+        with_clause = """
+            SELECT
+                mail_activity.res_id AS res_id,
+                min(date_deadline - (%%s AT TIME ZONE res_partner.tz)::timestamp) AS days_to_deadline
```

Comment:
res_partner.tz is not a required field,  `AT TIME ZONE NULL` returns null.

Then, I agree the odony's opinion about fields.Date.context_today

Reply:
Fixed by using ``COALESCE(res_partner.tz, 'UTC')``

https://github.com/odoo/odoo/compare/0c8e2340389f9807ba53d85c22645869c343d783..870609e1dcab086246d5958f7e5a9112d05932d0